src/Guesser/TypeGuesser.php

```diff
@@ -15,11 +15,30 @@
 
 use Doctrine\ORM\Mapping\ClassMetadata;
 use Sonata\AdminBundle\Model\ModelManagerInterface;
+use Sonata\AdminBundle\Templating\TemplateRegistry;
 use Symfony\Component\Form\Guess\Guess;
 use Symfony\Component\Form\Guess\TypeGuess;
 
 class TypeGuesser extends AbstractTypeGuesser
 {
+    /**
+     * This is a mapping between the old deprecated value we provided in the TypeGuesser
+     * and the value we should use to get the correct SonataAdminBundle template. Making the change
+     * directly in the TypeGuesser would be a BC-break if a user overrides the templates config.
+     *
+     * NEXT_MAJOR: remove this constant.
+     *
+     * @internal
+     */
+    public const DEPRECATED_TYPES = [
```

Comment:
There is no deprecation triggered in this PR. Can you link to the PR that adds the deprecation? I feel like I don't have all the context needed to understand the PR.

Reply:
This is related to https://github.com/sonata-project/SonataAdminBundle/pull/6123

Do you think something should be added ?
A deprecation message in the `private function getTemplate($type)` method ?

Reply:
My issue is, if I add a deprecation message, the user will received it every time because the Bundle is currently using the deprecated constant. And changing this could be considered as a BC-break.

Reply:
All the deprecations that can be solved by the user sound good to me :)

Reply:
Should be ok this way.